pkg/clusternetwork/config.go

```diff
@@ -35,20 +42,43 @@ func NetworkConfigurationFactory(oclient configclient.Interface, operatorClient
 	if err != nil {
 		return nil, errors.Wrap(err, "error getting cluster network type")
 	}
+
+	// retrieve serviceCIDR using cluster config required for cni configurations
+	serviceCIDR, err := getServiceNetworkCIDR(oclient)
+	if err != nil || serviceCIDR == "" {
+		return nil, errors.Wrap(err, "error getting service network CIDR")
+	}
+
 	switch network {
 	case ovnKubernetesNetwork:
 		return &ovnKubernetes{
 			networkType{
 				name:           network,
-				oclient:        oclient,
 				operatorClient: operatorClient,
 			},
+			NewClusterNetworkCfg(serviceCIDR),
 		}, nil
 	default:
 		return nil, errors.Errorf("%s : network type not supported", network)
 	}
 }
 
+// NewClusterNetworkCfg assigns a serviceCIDR value and
+// returns a pointer to the clusterNetworkCfg struct
+func NewClusterNetworkCfg(serviceCIDR string) *clusterNetworkCfg {
+	return &clusterNetworkCfg{
+		serviceCIDR: serviceCIDR,
+	}
+}
+
+// GetServiceCIDR returns the serviceCIDR string
+func (ovn *ovnKubernetes) GetServiceCIDR() (string, error) {
+	if ovn.clusterNetworkConfig.serviceCIDR == "" {
+		return "", errors.Errorf("received empty value for cluster serviceCIDR")
+	}
```

Comment:
You don't need to check for error here but you can check in `NewClusterNetworkCfg`.